src/_pytest/_code/code.py

```diff
@@ -984,6 +984,7 @@ class ReprTraceback(TerminalRepr):
     entrysep = "_ "
 
     def toterminal(self, tw: TerminalWriter) -> None:
+        tw.code_highlight = False
```

Comment:
was this left over from debugging? this doesn't quite feel right here

Reply:
No this was added purposely. The tests fail if code_highlight is set to true (which by default is it). I assumed all tests were written asserting to non highlighted code, so I put it here. Please guide me on this.

Reply:
I suspect this shouldn't happen here, and maybe we'll need to adjust some of the tests.  I'm not 100% certain though

Reply:
```
FAILED testing/code/test_excinfo.py::TestFormattedExcinfo::test_repr_traceback_with_invalid_cwd - Failed: nomatch: 'def entry():'
FAILED testing/code/test_excinfo.py::TestFormattedExcinfo::test_exc_chain_repr_without_traceback[cause] - Failed: exact match: 'ValueError: invalid value'
FAILED testing/code/test_excinfo.py::TestFormattedExcinfo::test_exc_chain_repr_without_traceback[context] - Failed: exact match: 'ValueError: invalid value'
FAILED testing/io/test_terminalwriter.py::test_code_highlight[no markup] - AssertionError: assert ['\x1b[94mass...[39;49;00m\n'] == ['assert 0\n']
FAILED testing/acceptance_test.py::TestGeneralUsage::test_root_conftest_syntax_error - Failed: nomatch: '*raise SyntaxError*'
FAILED testing/acceptance_test.py::TestGeneralUsage::test_early_hook_error_issue38_1 - Failed: nomatch: '*INTERNALERROR*def pytest_sessionstart():*'
FAILED testing/acceptance_test.py::TestGeneralUsage::test_better_reporting_on_conftest_load_failure - assert ["ImportError...med 'qwerty'"] == ["ImportError...med 'qwerty'"]
FAILED testing/test_conftest.py::test_conftest_exception_handling - assert 'raise ValueError()' in ["ImportError while loading conftest '/tmp/pytest-of-melvault/pytest-0/test_conftest_exception_h...
FAILED testing/test_junitxml.py::TestPython::test_xfailure_xpass_strict[xunit1] - AssertionError: assert {'message': '...1B[39;49;00m'} == {'message': '...eds to fail!'}
FAILED testing/test_junitxml.py::TestPython::test_xfailure_xpass_strict[xunit2] - AssertionError: assert {'message': '...1B[39;49;00m'} == {'message': '...eds to fail!'}
FAILED testing/test_reports.py::TestReportSerialization::test_chained_exceptions[TestReport] - assert "ValueError('value error')" in "\x1b[94mdef\x1b[39;49;00m \x1b[92mtest_a\x1b[39;49;00m():\n  ...
FAILED testing/test_reports.py::TestReportSerialization::test_chained_exceptions[CollectReport] - assert "ValueError('value error')" in "test_chained_exceptions.py:5: in test_a\n    foo()\ntest_c...
FAILED testing/test_resultlog.py::TestWithFunctionIntegration::test_log_test_outcomes - assert -1 != -1
FAILED testing/python/collect.py::TestModule::test_show_traceback_import_error[0] - Failed: nomatch: 'ImportError while importing test module*'
FAILED testing/python/collect.py::TestModule::test_show_traceback_import_error[1] - Failed: nomatch: 'ImportError while importing test module*'
FAILED testing/python/collect.py::TestModule::test_show_traceback_import_error[2] - Failed: nomatch: 'ImportError while importing test module*'
FAILED testing/python/collect.py::TestModule::test_show_traceback_import_error_unicode - Failed: nomatch: 'ImportError while importing test module*'
```
These are all the tests which failed with code_highlight set to `True`. I'm lost on what is the right way to do this. Please guide me here.

@asottile  @nicoddemus

Reply:
Hi @gdhameeja,

The problem was the conditional in `_highlight`: that should be an `or`, not an `and`. I've fixed it and pushed to your branch, hope you don't mind.